src/emulation_core/mips/line_info.rs

```diff
@@ -330,3 +333,317 @@ impl VisualDatapath for MipsDatapath {
         }
     }
 }
+
+pub fn visual_line_to_data(
```

Comment:
This seems identical to the implementation in visual_datapath/utils.rs, can one of them be removed?